dashboard/src/containers/Routes.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import { Redirect, Route, RouteComponentProps, Switch } from "react-router";
+import NotFound from "../components/NotFound";
+import AppList from "./AppListContainer";
+import AppNew from "./AppNewContainer";
+import AppUpgrade from "./AppUpgradeContainer";
+import AppView from "./AppViewContainer";
+import ChartList from "./ChartListContainer";
+import ChartView from "./ChartViewContainer";
+import ClassListContainer from "./ClassListContainer";
+import { ClassViewContainer } from "./ClassView";
+import FunctionListContainer from "./FunctionListContainer";
+import FunctionViewContainer from "./FunctionViewContainer";
+import InstanceListViewContainer from "./InstanceListViewContainer";
+import InstanceView from "./InstanceView";
+import LoginFormContainer from "./LoginFormContainer";
+import PrivateRouteContainer from "./PrivateRouteContainer";
+import RepoListContainer from "./RepoListContainer";
+import ServiceCatalogContainer from "./ServiceCatalogContainer";
+
+interface IChartInfoProps {
+  namespace: string;
+}
+
+class Routes extends React.Component<IChartInfoProps> {
+  public static exactRoutes: {
+    [route: string]: React.ComponentType<RouteComponentProps<any>> | React.ComponentType<any>;
+  } = {
+    // IMPORTANT: These routes should be added to the NGINX configuration
+    "/apps/ns/:namespace": AppList,
+    "/apps/ns/:namespace/:releaseName": AppView,
+    "/apps/ns/:namespace/new/:repo/:id/versions/:version": AppNew,
+    "/apps/ns/:namespace/upgrade/:releaseName": AppUpgrade,
+    "/charts": ChartList,
+    "/charts/:repo": ChartList,
+    "/charts/:repo/:id": ChartView,
+    "/charts/:repo/:id/versions/:version": ChartView,
+    "/config/brokers": ServiceCatalogContainer,
+    "/config/repos": RepoListContainer,
+    "/functions/ns/:namespace": FunctionListContainer,
+    "/functions/ns/:namespace/:name": FunctionViewContainer,
+    "/services/brokers/:brokerName/classes/:className": ClassViewContainer,
+    "/services/brokers/:brokerName/instances/ns/:namespace/:instanceName": InstanceView,
+    "/services/classes": ClassListContainer,
+    "/services/instances/ns/:namespace": InstanceListViewContainer,
+  };
+
+  public render() {
+    return (
+      <Switch>
+        <Route exact={true} path="/" render={this.rootNamespacedRedirect} />
+        <Route exact={true} path="/login" component={LoginFormContainer} />
+        {Object.keys(Routes.exactRoutes).map(route => (
+          <PrivateRouteContainer
+            key={route}
+            exact={true}
+            path={route}
+            component={Routes.exactRoutes[route]}
+          />
+        ))}
+        <Route path="*" component={NotFound} />
```

Comment:
Do you need to pass the `path`. The documentation seems to ignore it https://reacttraining.com/react-router/web/example/no-match

Also, you mentioned that we could redirect to /404, why do we need to redirect? Isn't better that the route stays the one that you tried to access?

Reply:
> Do you need to pass the path

I don't it was just to be explicit.

> Also, you mentioned that we could redirect to /404, why do we need to redirect? Isn't better that the route stays the one that you tried to access?

That was an alternative for returning a 404 in the status of the HTTP request and not having add all the routes I did. The idea was to configure only in nginx a route `/404` to return the error page instead of adding all the possible routes. For example, a request would:

 (request: `/foo`) -> nginx -> returns 200(OK) -> the react app redirects to `/404` -> nginx -> returns 404

In any case I think that is too confusing so I'd rather just return a `200` with the "not found" page content.